decisions/02-timeline-functionality.md

```diff
@@ -0,0 +1,69 @@
+# Timeline control for animated design functions
+
+## Status
+
+Proposed
+
+## Context
+
+When fine tweaking values for an animated design function developers might find
+themselves in a situation where they just a need a single frame of the animation
+for reference. They want to be able to update an input value or a constant in
+code and judge the visual effect of this change based on a single frame.
+
+We had this experience when using Mechanic to develop the title and credit
+sequences for a movie.
+
+Using the current setup there are ways of manually adding this behavior, but
+they require developers to work with their own drawloops, add the needed input
+and take care of the conditional rendering (animate vs. still frame) themselves.
+
+This document proposes a timeline input as part of mechanic core. It is intended
+as an opt-in developer feature that's most useful in purely frame based
+animations. It makes heavy use of the [proposed](https://github.com/designsystemsinternational/mechanic/pull/156) new [Animation API](https://github.com/designsystemsinternational/mechanic/pull/157).
+
+## Decision
+
+We propose a timeline input that developers can opt into for animated design
+functions. To do so they'd have to set the `showTimeline` property of their
+function's settings to `true`.
+
+By default the timeline renders with two additional number inputs for the min
+frame number and the max frame number. The timeline scrubber interpolates
+between these two values. This makes sure a developer can zoom in and zoom out
+of the animation area they need.
```

Comment:
Why should there be a min input and no setting to hardcode it? Is there a situation really where we need a min frame? isn't it always 0? Removing it also would help with the confusion of range export.

Reply:
Using `min` and `max` inputs allows a user to zoom to specifc ranges of frames if they want to.

WIth the hardcoded value it is assumed a user will always want the entire animation as the range of the timeline (thus `0` is implied as min value in that case).

Reply:
Interesting. I didn't get that implication from the setting of the `timelineMaxFrames` setting. Maybe it's a naming thing.